Failed tests (11540): TestStoreRangeRebalance

The following test appears to have failed:

[#11540](https://circleci.com/gh/cockroachdb/cockroach/11540):

```
I0116 09:40:13.744778 948 storage/replica.go:1683  gossiping system config from store 1, range 1, hash 9cc0e456982ebc2671b6a93a4bfa43dee8125261
I0116 09:40:13.748945 948 storage/replica.go:1624  gossiping cluster id "" from store 1, range 1
I0116 09:40:13.751560 948 storage/split_queue.go:97  splitting range=1 [/Min-/Max) at keys [/Table/11/0]
I0116 09:40:13.752041 948 kv/range_cache.go:128  lookup range descriptor: key=/Table/11/0
I0116 09:40:13.752201 948 kv/range_cache.go:128  lookup range descriptor: key=/Meta2/"\x93\x88"
panic: first Range is not present on any store in the multiTestContext. [recovered]
	panic: first Range is not present on any store in the multiTestContext.

goroutine 8597 [running]:
github.com/cockroachdb/cockroach/util/tracer.(*Trace).Finalize(0xc8204dd2d0)
	/go/src/github.com/cockroachdb/cockroach/util/tracer/tracer.go:148 +0x227
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).FirstRange(0xc8206da780, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:407 +0x3b5
github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor.func1(0xc8205dac00, 0xc8204d5870, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:152 +0x30c
github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor(0xc8205dac00, 0xc8204d5870, 0x3, 0x3, 0x0, 0x1, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:164 +0x2cf
github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor.func1(0xc8205dac00, 0xc8204d5410, 0x2, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:158 +0x475
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1c00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 1 [chan receive]:
testing.RunTests(0x1ccd2d8, 0x23fb200, 0xe0, 0xe0, 0x1)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820753f00, 0x1)
	/tmp/workdir/go/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820753f00)
	/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:37 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820753f00)
	/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:36 +0x34
main.main()
	github.com/cockroachdb/cockroach/storage/_test/_testmain.go:508 +0x20a

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/tmp/workdir/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x27431c0)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1029 +0x76
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 8586 [IO wait]:
net.runtime_pollWait(0x7fd44bb504e0, 0x72, 0x7fd44bb49050)
	/tmp/workdir/go/src/runtime/netpoll.go:157 +0x63
net.(*pollDesc).Wait(0xc8204dcfb0, 0x72, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:73 +0x56
net.(*pollDesc).WaitRead(0xc8204dcfb0, 0x0, 0x0)
	/tmp/workdir/go/src/net/fd_poll_runtime.go:78 +0x44
net.(*netFD).accept(0xc8204dcf50, 0x0, 0x7fd44bb51188, 0xc8204e1960)
	/tmp/workdir/go/src/net/fd_unix.go:408 +0x2f6
net.(*TCPListener).AcceptTCP(0xc8200c4a18, 0x61a030, 0x0, 0x0)
	/tmp/workdir/go/src/net/tcpsock_posix.go:254 +0x77
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc4d0, 0xc82052eae0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8588 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).processRaft.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1588 +0xb48
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204d4bf0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8591 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).startGossip.func2()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:645 +0x3ee
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1920)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8602 [select]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc8204aa7e0, 0xece2c06fd, 0x2cbd1c2d, 0x2742c40, 0xc8205da720, 0xc8204dc850, 0xc8204df2b0, 0x62787c)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:176 +0x641
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1.1(0xc8204df2b0, 0xc8205dad50)
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:214 +0xf6
github.com/cockroachdb/cockroach/storage.(*storeRangeSet).Visit(0xc8205daea0, 0xc820513240)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:221 +0x46b
github.com/cockroachdb/cockroach/storage.(*replicaScanner).scanLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/scanner.go:216 +0x2fe
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1cc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8585 [chan receive]:
github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open.func1(0xc8201dcff0)
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:123 +0x5b
created by github.com/cockroachdb/cockroach/storage/engine.(*RocksDB).Open
	/go/src/github.com/cockroachdb/cockroach/storage/engine/rocksdb.go:124 +0x581

goroutine 8590 [runnable]:
sync.runtime_Semacquire(0xc8204df2e4)
	/tmp/workdir/go/src/runtime/sema.go:43 +0x26
sync.(*Mutex).Lock(0xc8204df2e0)
	/tmp/workdir/go/src/sync/mutex.go:82 +0x1ee
github.com/cockroachdb/cockroach/storage.(*Replica).redirectOnOrAcquireLeaderLease(0xc8204df2b0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:467 +0x6f
github.com/cockroachdb/cockroach/storage.(*Replica).getLeaseForGossip.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/replica.go:1573 +0xa8
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunTask(0xc8204dc850, 0xc8206fd8f8, 0xc8ffffff00)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:116 +0x284
--
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e18e0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8582 [semacquire]:
sync.runtime_Semacquire(0xc8205295bc)
	/tmp/workdir/go/src/runtime/sema.go:43 +0x26
sync.(*WaitGroup).Wait(0xc8205295b0)
	/tmp/workdir/go/src/sync/waitgroup.go:126 +0x118
github.com/cockroachdb/cockroach/storage.(*Store).WaitForInit(0xc820529400)
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:606 +0x3e
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).addStore(0xc8206da780)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:487 +0xde9
github.com/cockroachdb/cockroach/storage_test.(*multiTestContext).Start(0xc8206da780, 0xc8205b9ef0, 0x4)
	/go/src/github.com/cockroachdb/cockroach/storage/client_test.go:235 +0x1468
--
testing.tRunner(0xc8205b9ef0, 0x23fc460)
	/tmp/workdir/go/src/testing/testing.go:456 +0xdd
created by testing.RunTests
	/tmp/workdir/go/src/testing/testing.go:561 +0xaa4

goroutine 8599 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1c60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8587 [chan receive]:
github.com/cockroachdb/cockroach/util.ListenAndServe.func2()
	/go/src/github.com/cockroachdb/cockroach/util/net.go:88 +0x7b
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc4d0, 0xc8204e1880)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8601 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1ca0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8589 [select]:
github.com/cockroachdb/cockroach/storage.(*Store).Start.func3()
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:566 +0x15d
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e18c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8584 [select]:
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).startStats(0xc82015a3c0)
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:214 +0xf1a
github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).(github.com/cockroachdb/cockroach/kv.startStats)-fm()
	/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:201 +0x2e
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc4d0, 0xc8203373f0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8583 [select]:
github.com/cockroachdb/cockroach/storage.(*StorePool).start.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/store_pool.go:213 +0x208
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc4d0, 0xc8204e10a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8598 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1c20)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8600 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1c80)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70

goroutine 8596 [select]:
github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:317 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc8204dc850, 0xc8204e1be0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:90 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	18.954s
=== RUN   TestBatchBasics
I0116 09:40:26.388708 1008 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I0116 09:40:26.399987 1008 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I0116 09:40:26.410058 1008 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_test.go
+++ b/storage/client_test.go
@@ -487,25 +487,28 @@ func (m *multiTestContext) addStore() {
 			}
 		}
 	}
-	if err := store.Start(stopper); err != nil {
-		m.t.Fatal(err)
-	}
-	store.WaitForInit()
 
 	// Add newly created objects to the multiTestContext's collections.
+	// (these must be populated before the store is started so that
+	// FirstRange() can find the sender)
 	m.mu.Lock()
-	defer m.mu.Unlock()
 	m.stores = append(m.stores, store)
 	m.stoppers = append(m.stoppers, stopper)
 	sender := storage.NewStores(clock)
 	sender.AddStore(store)
 	m.senders = append(m.senders, sender)
-	if err := gossipNodeDesc(m.gossip, nodeID); err != nil {
-		m.t.Fatal(err)
-	}
 	// Save the store identities for later so we can use them in
 	// replication operations even while the store is stopped.
 	m.idents = append(m.idents, store.Ident)
+	m.mu.Unlock()
+
+	if err := gossipNodeDesc(m.gossip, nodeID); err != nil {
+		m.t.Fatal(err)
+	}
+	if err := store.Start(stopper); err != nil {
+		m.t.Fatal(err)
+	}
+	store.WaitForInit()
 }
 
 // gossipNodeDesc adds the node descriptor to the gossip network.

